### DISCUSSION TITLE
Dependence of obtained value in website_settings on site id

### DISCUSSION THREAD

**Discussion by twin-elements** [pimcore/pimcore#17707] [Q&A / Support] [answered]
I wanted to set the “phone” setting in “website_settings” to different values depending on the siteid.
But every time the result is the **first** result in the database table and siteId is not considered at all.
Is there any way to do this?

```
{% if pimcore_site_is_request() %}
    {% set siteId = pimcore_site_current().id %}
{% else %}
    {% set siteId = 0 %}
{% endif %}

{% set phone = pimcore_website_config('phone',siteId) %}

<a href="tel:{{ phone | replace({' ': ''}) }}">{{ phone }}</a>
```
![image](https://github.com/user-attachments/assets/812af649-a0d2-46b8-b8f9-3cec5604d2c7)

**Comment by kingjia90** [comment, marked as answer]
The `pimcore_website_config` twig funciton doesn't have a way to set a Site, therefore `pimcore_website_config('phone',siteId)` is actually wrong
See also https://github.com/pimcore/pimcore/blob/68f198d32add713f2e4025b5dcb6414754133145/lib/Twig/Extension/WebsiteConfigExtension.php#L43

What you want to achieve might be done by passing a document of the give site by `contentDocument` to the request stack, see
https://github.com/pimcore/pimcore/blob/3ffaeac8d9f07f877ea4f469015be1774de447b9/lib/Config.php#L200-L206
but do not know how

**Reply by kingjia90** [reply]
Alternatively, if you have different sites, the phone number could be as an inheritable property from the root of the site.

**Reply by kingjia90** [reply]
This could be an improvement, to add support to pass a specific Site ID as a parameter to this Twig function, instead of automatically detecting the current site

**Reply by twin-elements** [reply]
Right. I thought the twig function pimcore_website_config('phone') was the same as WebsiteSetting::getByName('phone')?->getData().

It would make sense.
I will write my own twig function.
Thanks for your help.
